File: src/main/java/org/goafabric/calleeservice/controller/CalleeController.java
package org.goafabric.calleeservice.controller;

import org.goafabric.calleeservice.logic.CalleeLogic;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping(value = "callees", produces = MediaType.APPLICATION_JSON_VALUE)
public class CalleeController {
    private final CalleeLogic calleeLogic;

    public CalleeController(CalleeLogic calleeLogic) {
        this.calleeLogic = calleeLogic;
    }

    @GetMapping("sayMyName")
    public Callee sayMyName (@RequestParam("name") String name) {
        return calleeLogic.sayMyName(name);
    }

    @GetMapping("sayMyOtherName/{name}")
    public Callee sayMyOtherName(@PathVariable("name") String name) {
        return calleeLogic.sayMyOtherName(name);
    }

    @PostMapping(value = "save", consumes = MediaType.APPLICATION_JSON_VALUE)
    public Callee save(@RequestBody Callee callee) {
        return calleeLogic.save(callee);
    }

}

File: src/main/kotlin/org/goafabric/calleeservice/controller/CalleeController.kt
package org.goafabric.calleeservice.controller

import org.goafabric.calleeservice.controller.dto.Callee
import org.goafabric.calleeservice.logic.CalleeLogic
import org.springframework.http.MediaType.APPLICATION_JSON_VALUE
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping(value = ["callees"], produces = [APPLICATION_JSON_VALUE])
class CalleeController (
    private val calleeLogic : CalleeLogic) {

    @GetMapping("sayMyName")
    fun sayMyName (@RequestParam("name") name : String) : Callee {
        return calleeLogic.sayMyName(name)
    }

    @GetMapping("sayMyOtherName/{name}")
    fun sayMyOtherName (@PathVariable("name") name : String) : Callee {
        return calleeLogic.sayMyOtherName(name)
    }

    @PostMapping(value = ["save"], consumes = [APPLICATION_JSON_VALUE])
    fun save(@RequestBody callee: Callee): Callee? {
        return calleeLogic.save(callee)
    }

}